qa/L0_metrics/cpu_metrics_test.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/python
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import re
+import threading
+import time
+import unittest
+from collections import defaultdict
+
+import numpy as np
+import requests
+import tritonclient.http as httpclient
+
+CPU_UTILIZATION = "nv_cpu_utilization"
+CPU_USED_MEMORY = "nv_cpu_memory_used_bytes"
+CPU_TOTAL_MEMORY = "nv_cpu_memory_total_bytes"
+
+
+def get_metrics():
+    utilization_pattern = re.compile(rf"{CPU_UTILIZATION} (\d+\.?\d*)")
+    used_bytes_pattern = re.compile(rf"{CPU_USED_MEMORY} (\d+)")
+    total_bytes_pattern = re.compile(rf"{CPU_TOTAL_MEMORY} (\d+)")
+
+    r = requests.get("http://localhost:8002/metrics")
```

Comment:
Torn here, but as part of Windows GA we will need to obtain and test CPU metrics--so we will need to run this test. `localhost` is not a friendly term in our Windows CI. Instead we usually follow the convention: 
```python
_tritonserver_ipaddr = os.environ.get("TRITONSERVER_IPADDR", "localhost")
r = requests.get(f"http://{_tritonserver_ipaddr}:8002/metrics")
```
I'm torn because while it would be good to lay this groundwork early, we may also be shifting our Windows testing strategy in the near future.

Reply:
I have no issue adding a line if it makes it easier for windows later. Note that this is mostly stolen from `pinned_memory_metrics_test.py` which will have the same issues

Reply:
Right, that's another reason I'm torn--it'll obviously need to be fixed in other places later on as well. But I guess this could be an opportunity to lessen some of the tech debt. Your call.

Reply:
Updated python tests to use this convention.